microsoft/fluentui
# fix an issue with ChoiceGroup and focus rectangles

* [X] Code is up-to-date with the `master` branch
* [X] You've run `yarn change` locally


PR flow tips:
* [X] Try to start with a Draft PR
* [x] Once you're ready (ideally the pipeline is passing) promote your PR to Ready for Review. This step will auto-assign reviewers for your PR.

## Current Behavior

When `ChoiceGroup` is the child of an element that has focus clicking on an option will incorrectly draw a focus rectangle around the previously selected option. This is observable on the docs page for `ChoiceGroup` but not in an exported Codepen as described in #21193.

## New Behavior

#20859 added code to check if an element is a sibling of `ChoiceGroup` but it only tested that the element under consideration was not a descendant of `ChoiceGroup`. This change updates this code to also check that the element being tested is not an ancestor of `ChoiceGroup`.

### BEFORE
![Animated GIF showing the behavior before the change](https://user-images.githubusercontent.com/93940821/148620547-7404ae8a-2254-48f1-836a-3585a1b3f80c.gif)

### AFTER
![Animated GIF showing the behavior after the change](https://user-images.githubusercontent.com/93940821/148620576-622e7f9a-e38a-4a8e-98c2-b65183c153b6.gif)


## How to Test

1. Pull this branch locally
2. Run `yarn start`
3. Select "@fluentui/public-docsite"
4. Open the docsite in your browser (this should happen automatically)
    1. Under the "React" section click "Controls"
    2. On the next page click through Basic Controls > ChoiceGroup
    3. Scroll down an interact with any of the Usage examples

## Related Issue(s)

https://github.com/microsoft/fluentui/issues/21193

Fixes #21193
